train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -39,7 +52,13 @@ else if (referenceSpeed < -speedLimit) {
 
 	@Override
 	public void setJoystickPosition(int joystickPosition) {
-		this.step = joystickPosition;		
+		this.step = joystickPosition;
+		
+		if(this.timer == null) {
+			this.timer = new Timer();
+			this.timer.schedule(task,0, 1000);
```

Comment:
1000ms isn't frequent enough, please change it to 500ms!

Reply:
Resolved!